Sources/PointFree/Database.swift

```diff
@@ -2,88 +2,36 @@ import Either
 import Foundation
 import Prelude
 
-#if os(iOS)
-  enum PostgreSQL {
-    enum ConnInfo {
-      case basic(String, Int, String, String, String)
-    }
-    struct Database {
-      init(_ connInfo: ConnInfo) throws {
-      }
-      func makeConnection() throws -> Connection {
-        return Connection()
-      }
-    }
-    struct Connection {
-      func execute(_ query: String, _ representable: [NodeRepresentable] = []) throws -> Node {
-        return Node()
-      }
-    }
-    public typealias NodeRepresentable = Any
-    struct Node {
-      subscript(_: Int, _: String) -> Node? {
-        return nil
-      }
-      var int: Int?
-      var string: String?
-    }
+public struct Database {
+  var createSubscription: (StripeSubscription, User) -> EitherIO<Error, Prelude.Unit>
```

Comment:
I thought about bundling these functions in concrete protocols and uninhabited types, _e.g._:

``` swift
protocol Database {
  static func createSubscription: (StripeSubscription, User)
    -> EitherIO<Error, Prelude.Unit>
}

enum Postgres: Database {
  static func createSubscription: (StripeSubscription, User)
    -> EitherIO<Error, Prelude.Unit> { … }
}

enum MockDatabase: Database {
  static func createSubscription: (StripeSubscription, User)
    -> EitherIO<Error, Prelude.Unit> { … }
}

AppEnvironment.with(\.database .~ MockDatabase.self) { … }
```

But this doesn't allow for more granular stubbing in tests. By using structs, we get some nice lensing:

``` swift
AppEnvironment.with(\.database.migrate .~ { throwE(Database.Error()) }) { … }
```